cEP-0017.md

```diff
@@ -0,0 +1,176 @@
+   - The DistributionRequirement class needs to be split into different classes for each
+     package manager and call its method to introduct more consistency.
+4. We then proceed to install bear dependencies such as for `ClangCloneDetectionBear` by
+   running `install_bears` method on each dependency bear.
+
+### Here is the process for upgrading bears
+
+1. Check the bear name is installed using `PipRequirement` or `CondaRequirement`.
+2. Iterate through `Requirement` instances to make sure that the required version of the
+   linter is installed or upgrade the version if required.
+
+### Split `DistributionRequirement` for each distribution
+
+Specifically this will cover the following platforms:
+
+| Operating System | Tools      |
+| ---------------- | ---------- |
+| Debian           | apt-get    |
+| Ubuntu           | apt-get    |
+| CentOS           | yum        |
+| Fedora           | dnf        |
+| Arch Linux       | pacman     |
+| openSUSE         | zypper     |
+| Void Linux       | xbps       |
+| macOS            | brew       |
+| Windows          | Chocolatey |
+
+### Using features from depman
+
+Depman provides a way to install dependencies by specifying them through `requirements.yml`.
+Currently, only `pip` and `apt-get` are supported but by adding support for new installers we
+would be able to generate a `requirements.yml` specifying the bears and their dependencies.
+It would be useful to have `requirements.yml` in each project for the user that uses coala
+therefore, the cib should provide this additional option.
+
+### Other issues
+
+Allow `generate_package` to generate conda packages and move it from coala-bears to a different
+location.
+
+# Code Samples
+
+### Making use of conda packages in cib
+
+```python
+    def install_bears(bear_names_list):
+        bear_failed_list = []
+        for bear_name in bear_names_list:
+            if PipRequirement(bear_name).is_installed() or
+               CondaRequirement(bear_name).is_installed():
+                    print(bear_name + ' is already installed.')
+                    continue
+            if not PipRequirement(bear_name).install_package() or
+               not CondaRequirement(bear_name).install_package():
+                    install_requirements(bear_name)
+                    install_bear_dependencies(bear_name)
+                    print('DONE')
+            else:
+                print('Failed')
+                bears_failed_list.append(bear_name)
+        return bears_failed_list
+```
+
+### Installing bear dependencies
+
+```python
+    def install_bear_dependencies(bear_name):
+        package_object = importlib.import_module(
+            'coala' + package_name + '.' + package_name)
+        for bear in getattr(package_object, bear_name).BEAR_DEPS:
+            install_bears(bear)
+```
+
+### UI to ask permission for dependencies to be installed
+
+```python
+    yes = { 'yes' , 'y' ,  'ye' ,  '' }
```

Comment:
I think these are already defined in `coalib.misc.Constants` , or somewhere like that.